data_model/src/query/mod.rs

```diff
@@ -244,6 +260,69 @@ pub mod model {
     }
 }
 
+macro_rules! impl_query {
+    ($($ty:ty => $output:ty),+ $(,)?) => { $(
+        impl Query for $ty {
+            type Output = $output;
+
+            fn encode_as_query_box(&self) -> Vec<u8> {
+                QueryBoxRef::from(self).encode()
+            }
```

Comment:
If encoding is so easy now and used in only one place then maybe use that construction in that place and remove this method?

Reply:
it's just a question of what API do we want to expose. Notice that `QueryBoxRef` is private atm